src/main/groovy/net/neoforged/gradleutils/spotless/SpotlessUtilsExtension.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.spotless
+
+import com.diffplug.gradle.spotless.JavaExtension
+import com.diffplug.gradle.spotless.SpotlessExtension
+import com.diffplug.gradle.spotless.SpotlessTask
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+
+import javax.inject.Inject
+import java.nio.file.Files
+
+@CompileStatic
+abstract class SpotlessUtilsExtension {
+    final File configPath
+
+    @Inject
+    SpotlessUtilsExtension(Project project) {
+        configPath = new File(project.rootDir, '.gradle/formatter-config.xml')
+
+        if (!configPath.exists()) {
+            Files.createDirectories(configPath.toPath().parent)
+
+            try (
+                final input = SpotlessUtilsExtension.getResourceAsStream('/formatter-config.xml');
+                final output = Files.newOutputStream(configPath.toPath())
+            ) {
+                byte[] buffer = new byte[1024];
+                int len
+                while ((len = input.read(buffer)) > 0) {
+                    output.write(buffer, 0, len)
+                }
+            }
+        }
```

Comment:
Why are we doing this work here? This looks to duplicate the work done by `ExtractSpotlessConfiguration` (or rather, that task duplicates work that would be done by this).

Reply:
Unlike the task, this only does it if the file doesn't exist because the spotless plugin expects the configuration file to exist.

Reply:
What's the difference between the task and this block of code in the extension, then? 

Wouldn't this always run and check if the config file exists, thereby rendering the task essentially useless (since by the time the task runs, the extension already ran and extracted the config)?

Reply:
If this block of code does it every time, then every gradle refresh and invocation will cause a random file to be written.

Reply:
What do you mean by "random file"?

Also, this still doesn't answer the question of: what does the task do at execution time that the extension constructor hasn't already done by configuration time? If both write to the formatter config file, and the extension always run, wouldn't it mean the task always does nothing?

Reply:
> What do you mean by "random file"?
The file isn't always relevant to the task being invoked.

And to answer the other question, if the GU version is changed, the file wouldn't be updated by the extension, just by the task.

Reply:
To quell future questions on this regard, that detail should be included as a comment there (that the code block in the extension exists because otherwise Spotless would complain).

However, I do have a new question: can we move all the work that the extension constructor does currently to `configure(JavaExtension)`? So the extraction and task configuration only happens as Spotless itself is configured by this extension, and not when the extension is created.

Reply:
Not all of it, no, spotless configuration may be done after evaluation and other similar times, when task creation or configuration is not exactly ideal.  
But the config creation can be, sure.

Reply:
Fair point, and yeah that's better.